### PR TITLE
Add config_flow

### PR DESCRIPTION
**Pull request by joejulian**
This is still very much a WIP, but I would appreciate any feedback you might have.

### CODE REVIEW

File: custom_components/daikinskyport/daikinskyport.py [hunk 1]
```diff
@@ -3,12 +3,15 @@
...
 
-from requests.exceptions import RequestException
+import async_timeout
+
+from requests.exceptions import RequestException, ConnectionError, Timeout
 
 from .const import DAIKIN_PERCENT_MULTIPLIER
 
-logger = logging.getLogger('daikinskyport')
+_LOGGER = logging.getLogger('daikinskyport')
```
Comment (knightjoel) [inline custom_components/daikinskyport/daikinskyport.py:14]: Hi @joejulian,

I'm not very familiar with HA internals or proper coding conventions for HA components. Can you share why you changed the name of the logging object?

I also wonder, if it's not directly related to config flow, if this should be split to a separate PR.

File: custom_components/daikinskyport/daikinskyport.py [hunk 2]
```diff
@@ -143,40 +73,40 @@ def get_thermostats(self):
...
-            logger.debug("Error connecting to Daikin Skyport while attempting to get "
+            _LOGGER.debug("Error connecting to Daikin Skyport while attempting to get "
                         "thermostat data.  Refreshing tokens and trying again.")
             if self.refresh_tokens():
                 return self.get_thermostat_info(deviceid)
             else:
                 return None
 
-    def get_thermostat(self, index):
+    async def get_thermostat(self, index):
```
Comment (knightjoel) [inline custom_components/daikinskyport/daikinskyport.py:105]: Same thoughts here as I said with logging above: "Why?" and "should this be split into a separate PR?"

File: custom_components/daikinskyport/daikinskyport.py (hunk 1 above)
Comment (joejulian) [reply custom_components/daikinskyport/daikinskyport.py:14]: I'm learning. Comparing what's here with newer examples, I'm not yet sure what's needed and what's not. I figured I'd open this to get feedback and once I have a handle on what's going on, I can close it and split it up into single topic PRs.
